rollup.config.js

```diff
@@ -18,9 +18,7 @@ const typescriptPath = `${
 // the current build path
 const tsconfigOverride = {
   compilerOptions: {
-    paths: {
-      "*": ["*", path.join(process.cwd(), "*")]
-    }
+    rootDirs: ["js", path.join(process.cwd(), "gen")]
```

Comment:
I see.. so we're not rooted in the gen dir? What is the purpose of this?

Reply:
This instructs TypeScript to resolve relative paths as equals from these roots... it is specifically designed to deal with situations where you generate some stuff in another path, but want it treated like it were part of the static path.

Saying this out loud again though to try to get the paths in the types to make better sense, but in the end, I still couldn't get it to work, and I had to use the `/// <amd-module name="name"/>` for the modules we "care" about having sensible names.  So I may actually back this out and go back to the `paths` solution which put it in `gen`.  That is going to be far easier than re-writing the `tsconfig.generated.json` on the fly.